src/main/java/uk/gov/hmcts/darts/common/service/impl/RetrieveCoreObjectServiceImpl.java

```diff
@@ -125,13 +135,24 @@ public CourthouseEntity retrieveCourthouse(String courthouseName) {
     @Override
     public JudgeEntity retrieveOrCreateJudge(String judgeName) {
         Optional<JudgeEntity> foundJudge = judgeRepository.findByNameIgnoreCase(judgeName);
-        return foundJudge.orElseGet(() -> createJudge(judgeName));
+        return foundJudge.orElseGet(() -> judgeRepository.createJudge(judgeName));
     }
 
-    private JudgeEntity createJudge(String judgeName) {
-        JudgeEntity judge = new JudgeEntity();
-        judge.setName(judgeName);
-        judgeRepository.saveAndFlush(judge);
-        return judge;
+    @Override
+    public DefendantEntity retrieveOrCreateDefendant(String defendantName, CourtCaseEntity courtCase) {
+        Optional<DefendantEntity> foundJudge = defendantRepository.findByNameIgnoreCase(defendantName);
```

Comment:
this will find defendant thats linked to another case, we wouldn't want that. i believe the plan is just to create a new defendant for the new case. Same for defence and prosecutor

Reply:
ACCEPT